stdlib/int32.ml

```diff
@@ -66,3 +66,20 @@ type t = int32
 
 let compare (x: t) (y: t) = Pervasives.compare x y
 let equal (x: t) (y: t) = compare x y = 0
+
+let compare_unsigned n m =
+  compare (sub n min_int) (sub m min_int)
```

Comment:
Signed integer overflow is undefined, but you rely on a particular behaviour. Fishy-fishy.

A new C primitive would be a more robust solution.

Reply:
> Signed integer overflow is undefined

What???  Overflows are perfectly defined in OCaml.  Undefined arithmetic is for lesser languages.

Reply:
As a frequent user of lesser languages, I'm delighted to hear these great news! The manual is totally silent on this important property of integer operations, so I assumed overflows to be undefined (as I've seen no definitions). Perhaps this should be clarified in the manual?

Reply:
See http://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html, section "Integer arithmetic":
> Integers are 31 bits wide (or 63 bits on 64-bit processors). All operations are taken modulo 2^31 (or 2^63). They do not fail on overflow.

Reply:
Oh. I didn't grep the standard library headers. My mistake.

Overflows seem to be perfectly defined, after all (and I'm still delighted). Thanks for correcting me.